web/src/components/Korrel8rTopology.tsx

```diff
@@ -178,16 +179,13 @@ export const Korrel8rTopology: React.FC<{
       if (!newlySelectedNode) {
         return;
       }
-      const node = nodes.find((node) => node.id.startsWith(newlySelectedNode));
-      if (!node) {
-        return;
-      }
-      const url = nodes
-        .find((node) => node.id.startsWith(newlySelectedNode))
-        ?.data.korrel8rNode?.toURL();
-      history.push('/' + url);
+      const node = nodes.find((node) => node.id.startsWith(newlySelectedNode))?.data?.korrel8rNode;
```

Comment:
We can just leave the if statement and can avoid all the chaining after this point. It looks like the original one did the find twice which isn't needed, but the node variable should be fine to use

Reply:
Chaining is to get to the korrel8rNode inside the patternfly node - I cleaned it up a bit, please review.